arch/riscv/Kconfig

```diff
@@ -128,6 +128,24 @@ config GEN_IRQ_VECTOR_TABLE
 config NUM_IRQS
 	int
 
+config CODE_DATA_RELOCATION
+	bool "Support code/data section relocation"
+	depends on SOC_SERIES_RISCV32_IT8XXX2 \
+	  || SOC_RISCV32_LITEX_VEXRISCV \
+	  || SOC_SERIES_GD32VF103 \
+	  || SOC_SERIES_RISCV32_MIV \
+	  || SOC_SERIES_RISCV64_MIV \
+	  || SOC_SERIES_NEORV32 \
+	  || SOC_SERIES_RISCV_SIFIVE_FREEDOM \
+	  || SOC_SERIES_STARFIVE_JH71XX \
+	  || SOC_SERIES_RISCV_TELINK_B91 \
+	  || SOC_SERIES_RISCV_VIRT
+	help
+	  Enable support for relocating .text, data and .bss sections from specified
+	  files and placing them in a chosen memory region. Files to relocate and
+	  the target regions should be specified in CMakeLists.txt using
+	  zephyr_code_relocate().
```

Comment:
This is an exact duplicate of the one we have in `arch/arm/Kconfig` already, so why don't simply unify the two symbols in a general one for all the architectures supporting this?

Reply:
I didn't want to move the ARM version out of `arch/arm` since it has direct dependence on ARM specifics, but if we're adding a new symbol for "relocation is supported" as suggested then it makes sense to move it upward.